components/experimental/src/measure/mixed_measureunit.rs

```diff
@@ -0,0 +1,134 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::measure::{measureunit::MeasureUnit, parser::InvalidUnitError};
+
+use super::single_unit_vec::SingleUnitVec;
+
+/// The [`MixedMeasureUnit`] struct represents a CLDR mixed unit,
+/// which is a combination of one or more single units used together to express a measurement.
+///
+/// # Examples
+/// - `meter` - a special case of a mixed unit that contains only one single unit.
+/// - `foot-and-inch`
+/// - `kilometer-and-meter`
+///
+/// Note: Compound units such as `meter-per-second` or units with a constant denominator are not supported in mixed units.
+#[derive(Debug, Eq, Clone, PartialEq)]
+pub struct MixedMeasureUnit {
+    /// Contains the single mixed units.
+    pub(crate) mixed_units: SingleUnitVec,
+}
+
+impl MixedMeasureUnit {
+    /// Returns a slice of the mixed units contained within this mixed unit.
+    pub fn try_from_str(mixed_units_str: &str) -> Result<MixedMeasureUnit, InvalidUnitError> {
+        // '-and-' is the separator for the mixed units and it is allowed to appear in the start or end of the string.
+        let mixed_units_strs = mixed_units_str.split("-and-");
+        let mut mixed_units = SingleUnitVec::Zero;
+        for unit in mixed_units_strs {
+            let unit = MeasureUnit::try_from_str(unit)?;
+            let internal_single_units = unit.single_units();
+            if internal_single_units.len() > 1 {
+                return Err(InvalidUnitError);
+            }
+            if unit.constant_denominator != 0 {
+                return Err(InvalidUnitError);
+            }
+            let single_unit = match internal_single_units.first() {
+                Some(single_unit) => single_unit,
+                None => return Err(InvalidUnitError),
+            };
+            mixed_units.push(*single_unit);
+        }
+        Ok(MixedMeasureUnit { mixed_units })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::measure::provider::{
+        si_prefix::{Base, SiPrefix},
+        single_unit::SingleUnit,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_mixed_measure_unit_from_str() {
```

Comment:
issue: what does `kilometer-and-meter` test that `foot-and-inch` doesn't? what does `meter` test that `foot-and-inch` doesn't?

the reason why I think excessive testing like this is an issue is that these tests don't test the whole construct-format cycle, which is the actual API. instead, they test internal representation, which might need to be changed in the future, and then you need to update every single one of these test cases.

I'd prefer if you added the whole construct-format functionality in the same PR and tested that instead of the internal representation.